ext/mbstring/libmbfl/mbfl/mbfilter.c

```diff
@@ -855,13 +855,33 @@ mbfl_strpos(
 		needle_u8 = needle;
 	}
 
-	if (needle_u8->len < 1) {
-		result = (size_t) -8;
+	result = (size_t) -1;
+	if (haystack_u8->len < needle_u8->len) {
 		goto out;
 	}
 
-	result = (size_t) -1;
-	if (haystack_u8->len < needle_u8->len) {
+	if (needle_u8->len == 0) {
+		/* Out of bound offset */
+		if (
+			(offset > 0 && offset > mbfl_strlen(haystack_u8))
```

Comment:
mbfl_strlen() is very expensive, it should only be called once.

Reply:
I've made it store the value of mbfl_strlen, however I've reordered the rest meaning that it's going to be called every time. This is in part to improve the error reporting of these functions too in regards to the other PR as all of the functions don't inform of the errors the same way (or some just silently return false)